config/registration.bzl

```diff
@@ -4,13 +4,15 @@
 
 load("@crt//toolchains/gcc_arm_none_eabi:repository.bzl", "gcc_arm_none_eabi_repos")
 load("@crt//toolchains/lowrisc_rv32imcb:repository.bzl", "lowrisc_rv32imcb_repos")
+load("@crt//toolchains/cheri_llvm:repository.bzl", "cheri_llvm_repos")
 load("@crt//toolchains/gcc_mxe_mingw64:repository.bzl", "gcc_mxe_mingw64_repos")
 load("@crt//toolchains/cc65:repository.bzl", "cc65_repos")
 
 def crt_register_toolchains(
         arm = False,
         m6502 = False,
         riscv32 = False,
+        cheri_llvm = False,
```

Comment:
The CHERI LLVM toolchain can also generate riscv64 binaries, and even mips64 binaries (although this is almost never used nowadays) so I would hesitate to tie the name to riscv32.
The steps described in the README for building the archive specifically build libraries etc. that target riscv32+cheri, but they could be extended to also build the requirements for riscv64+cheri baremetal, mips64+cheri baremetal or even vanilla riscv32/64 and the built archive could then be used for any of those targets, so I think a more generic toolchain name is more appropriate.
Let me know your thoughts.

Reply:
What we're adding in this PR is a particular toolchain configuration that informs bazel about how to select and run the compiler to emit `riscv32` code with `cheri` features.

When/if we get to the point of having `riscv64+cheri` or `mips64+cheri` or `x86_64+cheri`, we'll need to add those as separate configurations (note that these configurations could indeed be backed by a single compiler and appropriate constraints to select the proper feature set per configuration).

Reply:
At the risk of annoying your half-to-death with requests to change the directory structure, it sounds like it might be appropriate to change `platforms/riscv32/cheri/...` to `platforms/cheri/riscv32/...`.  That would allow for the creation of a `riscv64`, `mips64` and whatever else under the `platforms/cheri` directory.

I still think the appropriate name here (in `crt_register_toolchains`) is a combination of `cheri` and the architecture name.  Even if you did eventually support a bunch of architectures with cheri options, downstream projects would probably only want to enable the configuration that was appropriate to their CPU architecture.

Reply:
OK, this makes sense. I've committed changes to rename the argument `cheri_llvm` to `cheri_riscv32` (the toolchain is still called `cheri_llvm`) and I've moved `platforms/riscv32/cheri` -> `platforms/cheri/riscv32`.